### PR TITLE
[Test Hackathon] Fix ForEach Testcases

### PR DESCRIPTION
**Pull request by arunans23**
## Purpose
Fix ForEach Testcases as part of Test Hackathon

### CODE REVIEW

File: integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/proxy-services/nested_foreach_iterate.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (chanikag) [inline integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/proxy-services/nested_foreach_iterate.xml]: Correct the license header

Comment (arunans23) [reply integration/mediation-tests/tests-mediator-1/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/proxy-services/nested_foreach_iterate.xml]: done

**Comment by a5anka** [comment]
`ForEachPropertyMediatorTestCase#sendRequest` method is duplicated in multiple places. Can we move it to a common place? 

Starting at line 390 of micro-integrator/integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForEachPropertiesTestCase.java
Starting at line 137 of micro-integrator/integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForEachPropertyMediatorTestCase.java
Starting at line 188 of micro-integrator/integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/NestedForEachPropertiesTestCase.java
Starting at line 138 of micro-integrator/integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/NestedForEachTestCase.java

    }

    private void sendRequest(String addUrl, String query) throws IOException {
        String charset = "UTF-8";
        URLConnection connection = new URL(addUrl).openConnection();
        connection.setDoOutput(true);
        connection.setRequestProperty("Accept-Charset", charset);
        connection.setRequestProperty("Content-Type", "application/xml;charset=" + charset);
        OutputStream output = null;
        try {
            output = connection.getOutputStream();
            output.write(query.getBytes(charset));
        } finally {
            if (output != null) {
                output.close();
            }
        }
        InputStream response = connection.getInputStream();
        if (response != null) {
            StringBuilder sb = new StringBuilder();
            byte[] bytes = new byte[1024];
            int len;
            while ((len = response.read(bytes)) != -1) {
                sb.append(new String(bytes, 0, len));
            }
            response.close();
        }
    }
}

**Comment by a5anka** [comment]
Found a 25 line (140 tokens) duplication in the following files: 
```
Starting at line 202 of micro-integrator/integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForEachPropertiesTestCase.java
Starting at line 333 of micro-integrator/integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForEachPropertiesTestCase.java
```
                carbonLogReader.checkForLog("2_in_originalpayload", 20)) {
            //fe : original payload while in foreach
            //in : original payload outside foreach
            String payload = message;
            String search = "<m0:checkPrice(.*)</m0:checkPrice>";
            Pattern pattern = Pattern.compile(search, Pattern.DOTALL);
            Matcher matcher = pattern.matcher(payload);
            boolean matchFound = matcher.find();

            assertTrue(matchFound, "checkPrice element not found. Instead found : " + payload);

            if (matchFound) {
                int start = matcher.start();
                int end = matcher.end();
                String quote = payload.substring(start, end);

                assertTrue(quote.contains("<m0:group>Group2</m0:group>"), "Group Element not found");
                assertTrue(quote.contains("<m0:code>IBM</m0:code>"), "IBM Element not found");
                assertTrue(quote.contains("<m0:code>WSO2</m0:code>"), "WSO2 Element not found");
                assertTrue(quote.contains("<m0:code>MSFT</m0:code>"), "MSTF Element not found");
                assertTrue(quote.contains("<m0:code>SUN</m0:code>"), "SUN Element not found");
            }
        }

        if (carbonLogReader.checkForLog("2_fe_group", 20) ||
```

**Comment by a5anka** [comment]
@arunans23 any updates on this?

**Comment by arunans23** [comment]
> @arunans23 any updates on this?

@a5anka I have done the changes in the local repo. But couldn't push it due to a git conflict issue. I will update the PR as soon as the issue is resolved.

**Comment by arunans23** [comment]
@asa

> @arunans23 any updates on this?

@a5anka Did the changes. Please review and merge.

**Pull request merged**